src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/items/ItemContainerContentsResourceHandler.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.wrappers.items;
+
+import java.util.Objects;
+import net.minecraft.core.NonNullList;
+import net.minecraft.core.component.DataComponentType;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.component.ItemContainerContents;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.handlers.IItemContext;
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.resources.ResourceStack;
+import net.neoforged.neoforge.transfer.transaction.Transaction;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+import net.neoforged.neoforge.transfer.transaction.TransactionManager;
+
+/**
+ * Wraps the vanilla ComponentData of {@link ItemContainerContents} to allow it to be used as a {@link net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler IResourceHandler}
+ */
+// A very verbose name, but likely the most accurate
+public class ItemContainerContentsResourceHandler implements IResourceHandler<ItemResource> {
+    protected final int size;
+    protected final DataComponentType<ItemContainerContents> componentType;
+    protected final IItemContext itemContext;
+
+    public ItemContainerContentsResourceHandler(IItemContext itemContext, DataComponentType<ItemContainerContents> componentType, int size) {
+        this.componentType = componentType;
+        this.itemContext = itemContext;
+        this.size = size;
+    }
+
+    public ItemContainerContents getContents() {
+        return itemContext.getResource().getOrDefault(componentType, ItemContainerContents.fromItems(NonNullList.withSize(size(), ItemStack.EMPTY)));
```

Comment:
We should just lazily initialize the default, rather than reinstantiate it ever single call. (Ideally only instantiating it if the component isn't present)